packages/db/src/models/granule.ts

```diff
@@ -37,6 +38,17 @@ export default class GranulePgModel extends BasePgModel<PostgresGranule, Postgre
       .merge()
       .returning('cumulus_id');
   }
+
+  async delete(
+    knexOrTransaction: Knex | Knex.Transaction,
+    granule: PostgresGranule
+  ) {
+    if (granule.published) {
```

Comment:
This logic already lives at the API endpoint handler level. This error will be thrown by the API before it even attempts the PG delete. So I'm not sure we need it here as well?

Reply:
Yeah we definitely don't need it here BUT the idea was to be extra cautious because removing a published granule from PG could be a problem. I _think_ this was based on an earlier conversation but I don't believe there's any more context than "It seemed like a good idea".

Reply:
How about this: let's move the check for `if granule.published` and throwing of the error into the deleteGranulesAndFiles handler and remove it from this model. Then, if bulk delete re-uses the same deleteGranulesAndFiles handler, which I hope it will, it will enforce the same check. 

It just feels like this kind of business logic doesn't quite belong in the database model and is a bit hidden here.

Reply:
Sure, that makes sense. I keep pushing to put more stuff on the PG models 😄 

https://github.com/nasa/cumulus/pull/2057/commits/61b9f7a2ec5b05feb58e1cb7d204afea20503c00